src/verify_cert.rs

```diff
@@ -19,12 +19,109 @@ use crate::{
     TrustAnchor,
 };
 
+/// Builds a RevocationOptions instance to control how revocation checking is performed.
+#[derive(Debug, Copy, Clone)]
+pub struct RevocationOptionsBuilder<'a> {
+    crls: &'a [&'a dyn CertRevocationList],
+
+    depth: RevocationCheckDepth,
+
+    status_requirement: UnknownStatusPolicy,
+}
+
+impl<'a> RevocationOptionsBuilder<'a> {
+    /// Create a builder that will perform revocation checking using the provided certificate
+    /// revocation lists (CRLs). At least one CRL must be provided.
+    ///
+    /// Use [build] to create a [RevocationOptions] instance.
+    ///
+    /// By default revocation checking will be performed on both the end-entity (leaf) certificate
+    /// and intermediate certificates. This can be customized using the [with_depth] method.
+    ///
+    /// By default revocation checking will fail if the revocation status of a certificate cannot
+    /// be determined. This can be customized using the [allow_unknown_status] method.
+    pub fn new(crls: &'a [&'a dyn CertRevocationList]) -> Result<Self, CrlsRequired> {
+        match crls.len() {
```

Comment:
So this probably goes against the intuition that you've built for my feedback, but I'd probably spell this as

```rust
if crls.is_empty() {
    return Err(CrlsRequired(()));
}

Ok(Self {
    ..
})
```

Why I wouldn't use the symmetric approach here: we can follow the linearization pattern here of early-returning the error case, and in this case there's no overhead of having get a usable value out of an `Option` or `Result` for the no-error path.

There also used to be a clippy lint (I think it's no longer enabled by default) that argues against using `len() == 0` in favor of using `is_empty()` on the grounds that this is more obvious. It may not have fired for this `match` setup, but I guess it still managed to imprint itself on my brain.

Whew, way too many words on silly little syntax details. Feel free to make of this what you want.